src/main/java/org/casbin/jcasbin/main/CoreEnforcer.java

```diff
@@ -344,14 +344,15 @@ public void buildRoleLinks() {
     }
 
     /**
-     * enforce decides whether a "subject" can access a "object" with
-     * the operation "action", input parameters are usually: (sub, obj, act).
+     * enforce use a custom matcher to decide whether a "subject" can access a "object" with the operation "action",
+     * input parameters are usually: (matcher, sub, obj, act), use model matcher by default when matcher is "" or null.
      *
-     * @param rvals the request needs to be mediated, usually an array
-     *              of strings, can be class instances if ABAC is used.
+     * @param matcher the custom matcher.
+     * @param rvals   the request needs to be mediated, usually an array
+     *                of strings, can be class instances if ABAC is used.
      * @return whether to allow the request.
      */
-    public boolean enforce(Object... rvals) {
+    private boolean enforce(String matcher, Object... rvals) {
```

Comment:
@shink why overload this function? Golang doesn't support overloading. And I want Java to follow the same function naming as Go, so even Java supports function overloading, we can choose not to use it actually.

Reply:
@hsluoyz There are two functions in Golang, [enforce](https://github.com/casbin/casbin/blob/c139c7702bd98bc8809a40127b3a9dbb937d3ea0/enforcer.go#L438) which is private and [Enforce](https://github.com/casbin/casbin/blob/c139c7702bd98bc8809a40127b3a9dbb937d3ea0/enforcer.go#L632) which is public. But the function name in java is usually lowercase, so I overload the `enforce` function.

Reply:
OK